cpp-client/Dockerfile

```diff
@@ -26,9 +26,41 @@ RUN set -eux; \
 #
 # Download and build dependent libraries.
 #
+# Be mindful about the order below:
+# 1. Some libraries have cross dependencies (eg, arrow depends on grpc).
+# 2. If something fails, the docker cache at that point gets invalidated;
+#    so it makes sense to put more brittle stuff later on.
+#
+RUN --mount=type=bind,source=build-dependencies.sh,target=/build-dependencies.sh \
+    set -eux; mkdir -p /cpp-client/deps; cd /cpp-client-deps; \
+    bash -c "/build-dependencies.sh --clean protobuf"
```

Comment:
bash should be the default, I don't think we need to specify it, we should be able to execute the script directly.